raiden/network/rpc/client.py

```diff
@@ -166,22 +167,30 @@ def geth_discover_next_available_nonce(web3: Web3, address: AddressHex) -> Nonce
     return web3.eth.getTransactionCount(address, "latest")
 
 
-def check_address_has_code(client: "JSONRPCClient", address: Address, contract_name: str = ""):
+def check_address_has_code(
+    client: "JSONRPCClient", address: Address, contract_name: str = "", expected_code: bytes = None
+):
     """ Checks that the given address contains code. """
     result = client.web3.eth.getCode(to_checksum_address(address), "latest")
 
-    if not result:
+    def formatted_contract_name():
         if contract_name:
-            formated_contract_name = f"[{contract_name}]: "
+            return f"[{contract_name}]: "
         else:
-            formated_contract_name = ""
+            return ""
 
+    if not result:
         raise AddressWithoutCode(
             "{}Address {} does not contain code".format(
-                formated_contract_name, to_checksum_address(address)
+                formatted_contract_name(), to_checksum_address(address)
```

Comment:
The extra function call here is not necessary, this can be simply `"[{contract_name}]:...`

Reply:
Well, it's a function now.  The reason is, 
* we need to compute this only in one of the two error cases
* but I didn't copy the string computing in two places

Reply:
But true, it's much easier if we can change the message.